conn.go

```diff
@@ -169,7 +177,16 @@ func createConn(nextConn net.Conn, flightHandler flightHandler, handshakeMessage
 	// Handle inbound
 	go c.inboundLoop()
 
-	<-c.handshakeCompleted
+	if c.connectTimeout.Seconds() > 0.0 {
```

Comment:
I would do `math.MaxInt64` if the user didn't specify a timeout. That way we can only have one code path.

Might be worth confirming it isn't expensive, but I always worry the complexity could hide bugs.

Reply:
Fixed :+1: